packages/edit-navigation/src/components/name-display/index.js

```diff
@@ -27,6 +27,7 @@ export default function NameDisplay() {
 						__( `Edit menu name: %s` ),
 						menuName
 					) }
+					className="edit-navigation-name-display__menu-name-button"
```

Comment:
If possible we should try to avoid adding classnames just for the tests. Usually the preference is to select elements by their visible text or aria-label. That helps ensure the UI is accessible. Most of the time it's possible to do this.

For this one, the aria label might be preferable as it has a bit more context to make it stand out from other buttons on the page. Using the aria-label for the selector has the added benefit that it asserts that the aria-label is correct:
```js
const menuNameButton = await page.waitForXPath( '//button[contains(@aria-label, "Edit menu name: newName" ) ]' );
expect( menuNameButton ).toBeTruthy();
const menuNamebuttonText = await menuNameButton.evaluate( ( element ) => element.innerText );
expect( menuNameButtonText ).toBe( 'newName' );
```